### PR TITLE
fishPlugins.done: init at 1.16.0

### PR DESCRIPTION
**Pull request by malob**
###### Motivation for this change

I use this plugin, and it's currently not in `nixpkgs`.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 119280` at 3024b377 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built successfully:</summary>
  <ul>
    <li>fishPlugins.done</li>
  </ul>
</details>
<details>
  <summary>3 suggestions:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md">missing-phase-hooks</a>

`buildPhase` should probably contain `runHook preBuild` and `runHook postBuild`.

Near pkgs/shells/fish/plugins/build-fish-plugin.nix:37:10:
```
   |
37 |   inherit unpackPhase configurePhase buildPhase;
   |          ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md">missing-phase-hooks</a>

`configurePhase` should probably contain `runHook preConfigure` and `runHook postConfigure`.

Near pkgs/shells/fish/plugins/build-fish-plugin.nix:37:10:
```
   |
37 |   inherit unpackPhase configurePhase buildPhase;
   |          ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md">missing-phase-hooks</a>

`checkPhase` should probably contain `runHook preCheck` and `runHook postCheck`.

Near pkgs/shells/fish/plugins/build-fish-plugin.nix:69:3:
```
   |
69 |   checkPhase = ''
   |   ^
```
</li>
  </ul>
</details>

**Comment by malob** [comment]
(cc @pacien)

File: pkgs/shells/fish/plugins/done.nix [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    sha256 = "ZlEbfF/6gmOOvQysPL8RYx42Q2oc/oT6VsqupsKGrSs=";
+  };
+
+  checkPlugins = [ fishtape ];
+  checkPhase = ''
+    fishtape test/done.fish
+  '';
+
+  meta = {
+    description = "Automatically receive notifications when long processes finish.";
```
Comment (SuperSandro2000) [inline pkgs/shells/fish/plugins/done.nix:20]:
```suggestion
    description = "Automatically receive notifications when long processes finish";
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by pacien** [comment]
Did you get this plugin to work?
I couldn't get any notification from it, but it might be an incompatibility with something else in my setup.

Also, note that a higher version has been tagged and released: https://github.com/franciscolourenco/done/tree/1.16.1

**Comment by malob** [comment]
> Did you get this plugin to work?

Yeah I've been using it for a while now on macOS with Kitty. I also just tested it in the Terminal app and it works there as well. It's been a while since I used my NixOS machine, but as far as I recall it worked on there as well.

> Also, note that a higher version has been tagged and released: https://github.com/franciscolourenco/done/tree/1.16.1

Cool, I've updated the PR to the latest version.

**Pull request merged**